src/mods/blockedPlayerSettings/index.js

```diff
@@ -21,60 +19,82 @@ function blockedPlayerSettings() {
 
-		// Wire up all the unblock buttons
-		Array.from(document.querySelectorAll('.js-unblock-player')).forEach($button => {
-			$button.addEventListener('click', clickEvent => {
-				const name = clickEvent.target.getAttribute('data-player-name');
-				player.unblockPlayer(name);
+	const customSettingsHTML = `
+		<h3 class="textprimary">Blocked players</h3>
+		<div class="settings uimod-settings">${blockedPlayersHTML}</div>
+		<p></p>
+		<div class="btn purp js-close-custom-settings">Close</div>
+	`;
+
+	const $customSettings = makeElement({
+		element: 'div',
+		class: 'menu panel-black js-custom-settings uimod-custom-window js-blocked-list',
+		content: customSettingsHTML,
+	});
+	document.body.appendChild($customSettings);
 
-				// Remove the blocked player from the list
-				Array.from(
-					document.querySelectorAll(`.js-custom-settings [data-player-name="${name}"]`),
-				).forEach($element => {
-					$element.parentNode.removeChild($element);
-				});
+	// Wire up all the unblock buttons
+	Array.from(document.querySelectorAll('.js-unblock-player')).forEach($button => {
+		$button.addEventListener('click', clickEvent => {
+			const name = clickEvent.target.getAttribute('data-player-name');
+			player.unblockPlayer(name);
+
+			// Remove the blocked player from the list
+			Array.from(
+				document.querySelectorAll(`.js-blocked-list [data-player-name="${name}"]`),
+			).forEach($element => {
+				$element.parentNode.removeChild($element);
 			});
 		});
-		// And the close button for our custom UI
-		document.querySelector('.js-close-custom-settings').addEventListener('click', () => {
-			const $customSettingsWindow = document.querySelector('.js-custom-settings');
-			$customSettingsWindow.parentNode.removeChild($customSettingsWindow);
-		});
 	});
+	// And the close button for our custom UI
+	document.querySelector('.js-close-custom-settings').addEventListener('click', hideBlockedList);
+}
+
+function hideBlockedList() {
+	const $customSettingsWindow = document.querySelector('.js-blocked-list');
+	$customSettingsWindow.parentNode.removeChild($customSettingsWindow);
 }
 
 export default {
 	name: 'Blocked Players List',
 	description: 'Allows you to view and remove blocked players from the Settings window',
-	run: ({ registerOnDomChange }) => {
+	run: ({ registerOnDomChange, registerOnStateChange }) => {
 		blockedPlayerSettings();
 
 		// If the settings window becomes visible/invisible, we want to update it
 		registerOnDomChange(blockedPlayerSettings);
+
+		registerOnStateChange((oldState, newState) => {
+			const oldBlockListPlayers = Object.keys(oldState.blockList);
+			// Find players in new state that aren't in old state
+			const newlyBlockedPlayers = Object.keys(newState.blockList).filter(
+				playerName => !oldBlockListPlayers.includes(playerName),
+			);
+
+			// A new player was blocked, reload UI if it's open already
+			if (newlyBlockedPlayers.length > 0) {
+				if (document.querySelector('.js-blocked-list')) {
```

Comment:
What do you think about tracking whether or not the blocked list window is opened in a `tempState`, rather than using `document.querySelector`? That way we're hammering the DOM less often, which is nice. Or would that be too much of a hassle to implement right now (I'm happy to do this too)?

I ask because we should always know when we create and when we close the blocked list DOM, because this mod controls that in its entirety.

Maybe the tempState structure would look something like
```
tempState.openWindows = {
blockedList: true,
friendsList: false,
}
```

Reply:
Implemented in latest commits, definitely prefer to handle it this way.